### ISSUE TITLE
3.0.0-8: Problem extending component styles 

### ISSUE THREAD

**Issue by gi-alec** [rebassjs/rebass#495]
Haven't had any luck extending styles. Here is a [gist of the relevant code](https://gist.github.com/gi-alec/392f7a0e2f87c7237fb69e120b8949e0).
### Debugging steps taken
- Updated all related packages to latest version 
(rebass, styled-components, babel-plugin-styled-components, gatsby-plugin-styled-components)
- Reset node_modules
`rm -rf node_modules && npm i`
- Reset browser cache

The code is pretty standard as far as I can tell. 
Cheers

**Comment by gi-alec** [comment]
Just made an isolated repro, and it works just fine. I'm having a hard time figuring what about my project is causing the issue. There are a lot of technologies playing together.

Let me know if you see anything obvious.

**Comment by jxnblk** [comment]
Yeah, I'm not sure what the culprit here is. I'd suggest trying to isolate the issue by trying styled-components without Rebass, and also trying the previous stable releases of both styled-components and rebass

**Comment by gi-alec** [comment]
I've done a lot of hacking. The issue only appears once extending Rebass Button.
`styled.button` for instance, is working as expected. `styled(Button)`, no luck. I'll open it up with the styled-components guys and see if they see an error.

**Comment by dlbnco** [comment]
I had the same problem. Downgraded to `2.3.4` and it worked.

**Comment by probablyup** [comment]
A fix for this was released in [v4.0.0-beta.8](https://github.com/styled-components/styled-components/releases/tag/v4.0.0-beta.8).

Long story short if two versions of hoist-non-react-statics are bundled together they poison each other.
